dbms/src/Flash/Pipeline/Schedule/Events/Event.h

```diff
@@ -95,6 +98,9 @@ class Event : public std::enable_shared_from_this<Event>
 
     std::atomic_int32_t unfinished_inputs{0};
 
+    // hold the tasks that ready to be scheduled.
+    std::vector<TaskPtr> tasks;
```

Comment:
tasks won't be modified by mutiple threads?

Reply:
Yes, `tasks ` only used in `Event::schedule()`.